### PR TITLE
[postponed/iceboxed] gas oracle

### PR DESCRIPTION
**Pull request by juniuszhou**
the pr implement a gas oracle, for the https://github.com/Sifchain/peggy/issues/52

### CODE REVIEW

File: cmd/ebrelayer/replay.go [hunk 1]
```diff
@@ -216,3 +216,64 @@ func RunListMissedCosmosEventCmd(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
```
Comment (banshee) [inline cmd/ebrelayer/replay.go:219]: Why is this code in replay.go?  Doesn't seem like it has anything to do with replay.

**Review by ElliotFriedman — CHANGES_REQUESTED** [review]

File: cmd/ebrelayer/replay.go (hunk 1 above)
Comment (juniuszhou) [reply cmd/ebrelayer/replay.go:219]: accept, will rename the source file.

File: cmd/ebrelayer/relayer/ethereum.go [hunk 2]
```diff
@@ -481,3 +481,66 @@ func (sub EthereumSub) handleEthereumEvent(events []types.EthereumEvent) error {
...
+				"ethereum block hash", newHead.Hash())
+
+			gasPrice, err := txs.GetGasPrice(sub.EthProvider, sub.SugaredLogger)
+			if err != nil {
+				sub.SugaredLogger.Errorw("failed to get gas price.",
+					errorMessageKey, err.Error())
+
+			} else {
+				// multiple with 150% to guarantee the speedy unpeg
+				gasPrice = gasPrice.Mul(gasPrice, big.NewInt(3))
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/relayer/ethereum.go:536]: will put it in keeper.

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 3]
```diff
@@ -159,3 +160,23 @@ func initRelayConfig(provider string, registry common.Address, event types.Event
 	}
 	return client, transactOptsAuth, target, nil
 }
+
+// GetGasPrice set up Ethereum client, validator's transaction auth, and the target contract's address
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:164]: accept.

**Comment by phips28** [comment]
Will this enable dynamic fees in the frontend? currentlty its 220$ to unpeg. Pretty insane high 🙈 
Is there an ETA on merging this?
Can I help?

**Comment by ElliotFriedman** [comment]
Do not merge, do not modify. This may be picked up at a later point in time after 42 and peggy 2.0.

**Pull request closed without merging**